WebContent/js/actions/editor.js

```diff
@@ -63,8 +63,25 @@ export function fetchDS(file) {
         dispatch(requestDSContent(file));
         const endpoint = `datasets/${encodeURIComponent(file)}/content`;
         let etag;
-        return atlasGet(endpoint, { })
+        return atlasGet(endpoint, {})
             .then(response => {
+                if (!response.ok) {
+                    return response.json().then(e => {
+                        let message;
+                        try {
+                            message = JSON.parse(e.message);
+                        } catch (err) {
+                            message = e.message;
+                        }
+
+                        if (message instanceof Array) {
```

Comment:
I think this is the only code path that isn't covered by a test? Could you add a test that returns an array as the API response please

Reply:
thanks @jordanCain , just added a relevant test case, using "first error message"  if its a json-string-array